torch/_refs/nn/functional/__init__.py

```diff
@@ -156,3 +157,45 @@ def softplus(
         rhs = refs.log1p(refs.exp(scaled_input))
 
     return refs.where(refs.gt(scaled_input, threshold), a, rhs)
+
+# Losses
+def _apply_loss_reduction(loss: TensorLikeType, reduction: str) -> TensorLikeType:
+    if reduction == 'sum':
+        return refs.sum(loss)
+    elif reduction == 'mean':
+        return refs.true_divide(refs.sum(loss), loss.numel())
+    else:
+        return loss
+
+def _check_reduction_value(reduction: str):
+    if reduction not in ('mean', 'sum', 'none'):
+        raise ValueError("{} is not a valid value for reduction".format(reduction))
+
+def margin_ranking_loss(input1: TensorLikeType, input2: TensorLikeType,
+                        target: TensorLikeType, margin=0.0, reduction='mean') -> TensorLikeType:
+    # loss_without_reduction = max(0, −target * (input1 − input2) + margin)
```

Comment:
I think it is helpful to have this for reference

Reply:
Elaborate on this comment explaining what this is a reference for